### PR TITLE
ASM-6387 ASM to support Urgent Fury vSAN RA

### PR DESCRIPTION
**Pull request by sushilrai**
Added support for boot_device "VSAN_AHCI" where Embedded SATA mode will be set to AHCIMode and SATA Disk 1 will be added as first disk in HDD Sequence. This will enable OS installation of first SATA Disk

### CODE REVIEW

File: lib/puppet/provider/importtemplatexml.rb [hunk 1]
```diff
@@ -184,7 +184,12 @@ def default_changes
       changes["partial"].deep_merge!("BIOS.Setup.1-1" => {"HddSeq" => "Disk.SDInternal.1-1"})
     end
 
-    if @boot_device =~ /VSAN/i
+    if @boot_device =~ /AHCI_VSAN/i
+      changes["partial"].deep_merge!("BIOS.Setup.1-1" => {"EmbSata" => "AhciMode"})
+      changes["partial"].deep_merge!("BIOS.Setup.1-1" => {"SecurityFreezeLock" => "Disabled"})
+      changes["partial"].deep_merge!("BIOS.Setup.1-1" => {"WriteCache" => "Disabled"})
+      changes["partial"].deep_merge!("BIOS.Setup.1-1" => {"HddSeq" => "Disk.SATAEmbedded.A-1"})
```
Comment (gavin-scott) [inline lib/puppet/provider/importtemplatexml.rb]: Is it always Disk.SATAEmbedded.A-1? @AlexDron does your filtering for AHCI_VSAN guarantee that the server has this disk?

Comment (AlexDron) [inline lib/puppet/provider/importtemplatexml.rb]: Yes, AHCI_VSAN is a target boot type for this kind of filtering

Comment (gavin-scott) [inline lib/puppet/provider/importtemplatexml.rb]: I guess my question was -- when you filter for AHCI_VSAN are we guaranteed that the server has a Disk.SATAEmbedded.A-1? I think the answer is no based on when I reviewed the filtering PR. If we're not guaranteed of that then we need to either add that as part of filtering or add code here that enumerates the disks and gets the first one present.

**Comment by sushilrai** [comment]
@gavin-scott  I am working on the logic for getting the SATA disk enumeration. Have sent query to the LC team on this.

The tests are failing due to rbvmomi package require. I dont see any reference of rbvmomi in idrac module. Even my changes are very limited to the boot order and havent added changes which add the dependency for rbvmomi.
Can you please take a look at it.

**Comment by gavin-scott** [comment]
@sushilrai if you rebase the tests should be fixed. The rbvmomi issue was fixed by #208 which removed the need for the idrac module to use asm-deployer code. (that's what was pulling in rbvmomi)

**Comment by sushilrai** [comment]
@gavin-scott  this PR is merged to master branch but sill the changes are missing the 8.2.0 branch. Are we planning to port it back to 8.2.0. Need this PR to complete VSAN changes for Urgent Fury work

**Comment by gavin-scott** [comment]
Tests are failing due to asm-deployer dependency that we can't get rid of for 8.2.0. I'll go ahead and merge.

**Pull request merged**